### ISSUE TITLE
assert((areEqual != null) == (getHashCode != null)) should be inverted

### ISSUE THREAD

**Issue by micimize** [marcelgarus/list_diff#3]
As far as I can tell this checks the opposite of what you wan to check
```dart
('areEqual' != null) == ('getHashCode' != null) ///=> `true`, does not throw
('areEqual' != null) == (null != null) ///=> `false`, throws
```
https://github.com/marcelgarus/list_diff/blob/649172c791351228dbd04d72ce18572ead243661/lib/list_diff.dart#L84-L87

**Comment by MarcelGarus** [comment]
You're completely right, I'll fix that soonish.

**Comment by MarcelGarus** [comment]
Actually, you're supposed to either provide both methods or none at all. So I just fixed the assert message. The new version is published as 1.3.1 on pub.dev.
